### PR TITLE
xtensa-build-zephyr: parse SOF versions from new versions.json instead of generated sof_versions.h

### PR DESCRIPTION
**Pull request by marc-hb**
\+ move some rimage-related code around to prepare for the switch to `west build` (1st part of successfully tested  draft TEST PR #7595)

No functional change.

One of the last parts for this v2.6 requirement:
- #6917

### CODE REVIEW

**Comment by marc-hb** [comment]
Only one test in https://sof-ci.01.org/sofpr/PR7661/build8213/devicetest/index.html?model=MTLP_RVP_SDW&testcase=check-runtime-pm-status failed like this. This failure looks new but it can't be caused by this PR, it's unrelated:

```
2023-05-20 00:01:33 UTC [REMOTE_COMMAND] sof-tplgreader.py /usr/lib/firmware/intel/sof-ace-tplg/sof-mtl-rt711-4ch.tplg -f 'type:any & ~pcm:HDMI' -b ' pcm:HDA Digital' -s 0 -e
2023-05-20 00:01:33 UTC [REMOTE_INFO] ===== Iteration 1 of 3 for DMIC =====
2023-05-20 00:01:33 UTC [REMOTE_COMMAND] arecord -D hw:0,10 -r 48000 -c 4 -f S32_LE /dev/null -q
2023-05-20 00:01:36 UTC [REMOTE_INFO] runtime status: active
2023-05-20 00:01:36 UTC [REMOTE_COMMAND] kill process: kill -9 30098
2023-05-20 00:01:36 UTC [REMOTE_INFO] arecord killed
2023-05-20 00:01:36 UTC [REMOTE_INFO] wait dsp power status to become suspended
2023-05-20 00:01:48 UTC [REMOTE_ERROR] dsp is not suspended after 10s, end test
2023-05-20 00:01:48 UTC [REMOTE_ERROR] Starting func_exit_handler(), exit status=1, FUNCNAME stack:
2023-05-20 00:01:48 UTC [REMOTE_ERROR]  die()  @  /home/ubuntu/sof-test/test-case/../case-lib/lib.sh
2023-05-20 00:01:48 UTC [REMOTE_ERROR]  func_check_dsp_status()  @  /home/ubuntu/sof-test/test-case/check-runtime-pm-status.sh:46
2023-05-20 00:01:48 UTC [REMOTE_ERROR]  main()  @  /home/ubuntu/sof-test/test-case/check-runtime-pm-status.sh:112
2023-05-20 00:01:48 UTC [REMOTE_INFO] pkill -TERM -f mtrace-reader.py
2023-05-20 00:01:48 UTC [REMOTE_INFO] ktime=481 sof-test PID=29904: ending
2023-05-20 00:01:48 UTC [REMOTE_INFO] Test Result: FAIL!

```

https://sof-ci.01.org/sofpr/PR7661/build8214/devicetest and everything else are all green.

**Review by paulstelian97 — APPROVED** [review]

File: scripts/xtensa-build-zephyr.py [hunk 1]
```diff
@@ -443,27 +444,26 @@ def west_update():
...
-		for hline in hfile:
-			words = hline.split()
-			if words[0] == '#define':
-				versions[words[1]] = words[2]
-	sof_fw_version = versions['SOF_MAJOR'] + '.' + versions['SOF_MINOR'] + '.' + \
-		      versions['SOF_MICRO']
-
+	with open(SOF_TOP / "versions.json") as versions_file:
+		versions = json.load(versions_file)
+		sof_micro = versions['SOF'].get('MICRO')
```
Comment (aiChaoSONG) [inline scripts/xtensa-build-zephyr.py:461]: The get here allow a default value if the value we want to get doesn't exist, why not just simply

`sof_micro = versions[''SOF].get('MICRO', '0')`

Comment (marc-hb) [reply scripts/xtensa-build-zephyr.py]: Done

**Review by aborisovich — APPROVED** [review]
OK I won't be blocking this because there are dependencies to Zephyr as @marc-hb described in the feature request. Python code looks ok. We just need to find the time to improve this solution.

**Pull request merged**